openshift-knative-operator/pkg/serving/extension.go

```diff
@@ -165,11 +165,16 @@ func (e *extension) fetchLoggingHost(ctx context.Context) string {
 // Also handles the (buggy) case, where all Ingresses are disabled.
 // See https://github.com/knative/operator/issues/568.
 func defaultToKourier(ks *v1alpha1.KnativeServing) {
-	if ks.Spec.Ingress == nil || (!ks.Spec.Ingress.Istio.Enabled && !ks.Spec.Ingress.Kourier.Enabled && !ks.Spec.Ingress.Contour.Enabled) {
+	if ks.Spec.Ingress == nil {
 		ks.Spec.Ingress = &v1alpha1.IngressConfigs{
 			Kourier: v1alpha1.KourierIngressConfiguration{
 				Enabled: true,
 			},
 		}
+		return
+	}
+
+	if !ks.Spec.Ingress.Istio.Enabled && !ks.Spec.Ingress.Kourier.Enabled && !ks.Spec.Ingress.Contour.Enabled {
```

Comment:
```suggestion
	if !ks.Spec.Ingress.Istio.Enabled && !ks.Spec.Ingress.Kourier.Enabled {
```

I wonder that we should enable Kourier regardless of the Contour setting. But current behavior is also fine as users can realize it by the deploy error. What do you think?

Reply:
You mean we should bail out if a user tries to enable contour? We could do that yeah, though I'd like to do that separately as we'd ideally also add webhook validation to that then.

Reply:
Sure, webhook validation is ideal. Let's leave it as it is.